src/product_representations.jl

```diff
@@ -229,20 +244,26 @@ struct ProductRepr{TM<:Tuple}
 end
 
 ProductRepr(points...) = ProductRepr{typeof(points)}(points)
-eltype(x::ProductRepr) = eltype(Tuple{map(eltype, x.parts)...})
-similar(x::ProductRepr) = ProductRepr(map(similar, x.parts)...)
-similar(x::ProductRepr, ::Type{T}) where T = ProductRepr(map(t -> similar(t, T), x.parts)...)
+eltype(x::ProductRepr) = eltype(Tuple{map(eltype, submanifold_components(x))...})
+similar(x::ProductRepr) = ProductRepr(map(similar, submanifold_components(x))...)
+function similar(x::ProductRepr, ::Type{T}) where {T}
+    return ProductRepr(map(t -> similar(t, T), submanifold_components(x))...)
+end
 
 function copyto!(x::ProductRepr, y::ProductRepr)
-    map(copyto!, x.parts, y.parts)
+    map(copyto!, submanifold_components(x), submanifold_components(y))
     return x
 end
 
-function submanifold_component(x::ProductRepr, i::Integer)
-    return x.parts[i]
-end
-
 (+)(v1::ProductRepr, v2::ProductRepr) = ProductRepr(map(+, v1.parts, v2.parts)...)
 (-)(v1::ProductRepr, v2::ProductRepr) = ProductRepr(map(-, v1.parts, v2.parts)...)
-(-)(v::ProductRepr) = ProductRepr(map(-, v.parts))
-(*)(a::Number, v::ProductRepr) = ProductRepr(map(t -> a*t, v.parts))
+(-)(v::ProductRepr) = ProductRepr(map(-, submanifold_components(v)))
+(*)(a::Number, v::ProductRepr) = ProductRepr(map(t -> a*t, submanifold_components(v)))
+
+function isapprox(x::ProductRepr, y::ProductRepr; kwargs...)
```

Comment:
Are you sure this is needed? `ProductRepr` should generally be used in point or tangent vector `isapprox` for a specific manifold.

Reply:
Good point. Fixed.